src/equations/compressible_euler_2d.jl

```diff
@@ -1263,4 +1263,43 @@ end
 end
 
 
+# Calculate the vorticity on a single node using the derivative matrix from the polynomial basis of
+# a DGSEM solver.
+# !!! Note that this only works for DGSEM solvers on a TreeMesh !!!
+# `u` is the solution on the whole domain.
+# This function is used for calculating acoustic source terms for coupled Euler-acoustics
+# simulations.
+function calc_vorticity_node(u, equations::CompressibleEulerEquations2D, dgsem, cache, i, j, element)
+  @unpack derivative_matrix = dgsem.basis
```

Comment:
`dg::DGSEM` instead of `dgsem`?

Reply:
Unfortunately, this does not work since `equations.jl` is included before `solvers.jl` and they can't be reordered because `DGMulti` flux differencing depends on `CompressibleEulerEquations2D`.

Reply:
See above (https://github.com/trixi-framework/Trixi.jl/pull/712#discussion_r709166655)